meson.build

```diff
@@ -0,0 +1,1554 @@
+    'use': false,
+}
+if not glibc_thread_name['use']
+    osx_thread_name += {'use': cc.compiles(osx_thread_name['statement'], dependencies: pthreads, name: 'osx-thread-name check')}
+endif
+
+bsd_thread_name = {
+    'name': 'bsd-thread-name',
+    'statement': run_command('cat', 'waftools/fragments/bsd_thread_name.c').stdout(),
+    'use': false,
+}
+if not glibc_thread_name['use'] and not osx_thread_name['use']
+    bsd_thread_name += {'use': cc.compiles(bsd_thread_name['statement'], dependencies: pthreads, name: 'bsd-thread-name check')}
+endif
+
+bsd_fstatfs = {
+    'name': 'bsd-fstatfs',
+    'statement': run_command('cat', 'waftools/fragments/bsd_fstatfs.c').stdout(),
+    'use': false,
+}
+bsd_fstatfs += {'use': cc.compiles(bsd_fstatfs['statement'], name: 'bsd-fstatfs check')}
+
+linux_fstatfs = {
+    'name': 'linux-fstatfs',
+    'statement': run_command('cat', 'waftools/fragments/linux_fstatfs.c').stdout(),
+    'use': false,
+}
+linux_fstatfs += {'use': cc.compiles(linux_fstatfs['statement'], name: 'linux-fstatfs check')}
+
+
+# misc pkg-config dependencies
+cdda = dependency('libcdio_paranoia', required: get_option('cdda'))
+if cdda.found()
+    dependencies += cdda
+    sources += 'stream/stream_cdda.c'
+endif
+
+dvdnav = dependency('dvdnav', version: '>= 4.2.0', required: get_option('dvdnav'))
+dvdread = dependency('dvdread', version: '>= 4.1.0', required: get_option('dvdnav'))
+if dvdnav.found() and dvdread.found()
+    dependencies += [dvdnav, dvdread]
+    sources += 'stream/stream_dvdnav.c'
+endif
+
+javascript = dependency('mujs', version: '>= 1.0.0', required: get_option('javascript'))
+if javascript.found()
+    dependencies += javascript
+    sources += 'player/javascript.c'
+endif
+
+lcms2 = dependency('lcms2', version: '>= 2.6', required: get_option('lcms2'))
+if lcms2.found()
+    dependencies += lcms2
+endif
+
+libarchive = dependency('libarchive', version: '>= 3.4.0', required: get_option('libarchive'))
+if libarchive.found()
+    dependencies += libarchive
+    sources += ['demux/demux_libarchive.c',
+                'stream/stream_libarchive.c']
+endif
+
+libavdevice = dependency('libavdevice', version: '>= 57.0.0', required: get_option('libavdevice'))
+if libavdevice.found()
+    dependencies += libavdevice
+endif
+
+libbluray = dependency('libbluray', version: '>= 0.3.0', required: get_option('libbluray'))
+if libbluray.found()
+    dependencies += libbluray
+    sources += 'stream/stream_bluray.c'
+endif
+
+lua_version = [['52', '>= 5.2.0, < 5.3.0'],
+               ['lua52', '>= 5.2.0'],
+               ['lua5.2', '>= 5.2.0'],
+               ['lua-5.2', '>= 5.2.0'],
+               ['luajit', '>= 2.0.0'],
+               ['51', '>= 5.1.0, < 5.2.0'],
+               ['lua51', '>= 5.1.0'],
+               ['lua5.1', '>= 5.1.0'],
+               ['lua-5.1', '>= 5.1.0']]
+foreach version : lua_version
+    lua = dependency(version[0], version: version[1], required: get_option('lua'))
```

Comment:
When option `-Dlua=enabled` is passed, it will never reach `['luajit', '>= 2.0.0']`. Maybe set `required: false` to tell meson not to error out while iterating over list?

Reply:
Good catch.